### ISSUE TITLE
Gifs get corrupted

### ISSUE THREAD

**Issue by felixfbecker** [Netflix/pollyjs#346]
## Description

When using `record` mode, gifs show up for me normally (using Puppeteer). But in `replay` mode, gifs are always corrupted.

This is reproducable on any page with gifs, e.g. https://news.ycombinator.com/ (the upvote arrows)

### Shareable Source

In the HAR, I can see that the gif is saved as a string - intuitively, I would guess this is the problem, and they should be Base64 encoded?

```json
"response": {
          "bodySize": 100,
          "content": {
            "mimeType": "image/gif",
            "size": 100,
            "text": "GIF89a\u0012\u0000\u0012\u0000�\u0000\u0000�����W��-��s����f\u0000\u0000\u0000\u0000\u0000\u0000\u0000!�\u0004\u0000\u0000\u0000\u0000\u0000,\u0000\u0000\u0000\u0000\u0012\u0000\u0012\u0000\u0000\u0003)X���0�9\t\u0011��A��D\u0000\n@PA�\u0005Ta�m�uq��w��{��@I\u0002\u0000;"
          },
```

### Error Message & Stack Trace

None

### Config

Copy the config used to setup the Polly instance:

```js
polly = new Polly('Recording', {
	recordFailedRequests: true,
	adapters: [PuppeteerAdapter],
	adapterOptions: {
		puppeteer: { page },
	},
	persister: FSPersister,
	persisterOptions: {
		fs: {
			recordingsDir: path.resolve(__dirname, 'recordings'),
		},
	},
})
```

### Dependencies

Copy the @pollyjs dependencies from `package.json`:

```json
{
		"@pollyjs/adapter-puppeteer": "^4.3.0",
		"@pollyjs/core": "^4.3.0",
		"@pollyjs/persister-fs": "^4.3.0",
}
```

## Environment

```
Node.js v14.4.0
darwin 19.5.0
```

**Comment by offirgolan** [comment]
Unfortunately, the puppeteer adapter doesn't support recording binary data as of right now.

**Comment by felixfbecker** [comment]
Interesting. Is that a technical limitation of Puppeteer's API? Or just something not handled in the adapter?
This makes Polly not useable for screenshot testing unfortunately since many webpages contain images :/

**Comment by offirgolan** [comment]
No, not a limitation on the Puppeteer side but it just wasn't implemented on this specific adapter. We recently finished adding binary data handling for the rest of the adapters so we will need to do the same thing here.

**Comment by felixfbecker** [comment]
@offirgolan is this planned to be implemented any time soon? Any pointers how it needs to be implemented?
